File: OOP based implmentation/OOP based implmentation/Project/test/mcqs.java
package test;

import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;
import java.util.Scanner;

public class mcqs {
    public static Scanner sc = new Scanner(System.in);

    public static float giveTest() {
        int score = 0;
        int totalQue = 0;       
        try (BufferedReader reader = new BufferedReader(new FileReader("src/resources/dsa_mcq_placement_test.txt"))) {
            String line;
            String ans = "";

            while ((line = reader.readLine()) != null) { 
                if (!line.isEmpty()) {
                    if (line.startsWith("Answer:")) {
                        ans = line.split(": ")[1].trim();
                        System.out.print("Give your answer: ");
                        String answer = sc.nextLine();

                        if (answer.equalsIgnoreCase(ans)) {
                            score++;
                        }
                        totalQue++;
                    } else {
                        System.out.println(line);
                    }
                }
            }

        } catch (IOException e) {
            System.out.println("An error occurred while reading the file.");
            e.printStackTrace();
        }
        
        return totalQue > 0 ? (score * 100.0f / totalQue) : 0;
    }
}

File: OOP based implmentation/OOP based implmentation/Project/test/mcqs.cpp
#include <iostream>
#include <fstream>
#include <string>
#include "mcqs.h"

using namespace std;

float MCQs::giveTest()
{
        int score = 0;
        int totalQue = 0;

        ifstream reader("resources/dsa_mcq_placement_test.txt");
        if (!reader.is_open())
        {
            cout << "An error occurred while reading the file." << endl;
            return 0.0f;
        }

        string line;
        string ans = "";

        while (getline(reader, line))
        {
            if (!line.empty())
            {
                if (line.rfind("Answer:", 0) == 0)
                {
                    ans = line.substr(line.find(":") + 1);
                    size_t first = ans.find_first_not_of(" \t\n\r");
                    size_t last = ans.find_last_not_of(" \t\n\r");
                    if (string::npos != first && string::npos != last)
                    {
                        ans = ans.substr(first, (last - first + 1));
                    }
                    cout << "Give your answer: ";
                    string answer;
                    cin >> answer;

                    if (answer == ans)
                    {
                        score++;
                    }
                    totalQue++;
                }
                else
                {
                    cout << line << endl;
                }
            }
        }
        reader.close();

        return totalQue > 0 ? (static_cast<float>(score) * 100.0f / totalQue) : 0.0f;
}